cdap-app-fabric/src/main/java/co/cask/cdap/test/internal/AppFabricClient.java

```diff
@@ -69,60 +77,88 @@
     .create();
 
   private final AppFabricHttpHandler httpHandler;
-  private final ServiceHttpHandler serviceHttpHandler;
   private final LocationFactory locationFactory;
-
-  public AppFabricClient(AppFabricHttpHandler httpHandler, ServiceHttpHandler serviceHttpHandler,
-                         LocationFactory locationFactory) {
+  private final AppLifecycleHttpHandler appLifecycleHttpHandler;
+  private final ProgramLifecycleHttpHandler programLifecycleHttpHandler;
+  private final NamespaceHttpHandler namespaceHttpHandler;
+  private final NamespaceAdmin namespaceAdmin;
+  private final StreamAdmin streamAdmin;
+  private final StreamMetaStore streamMetaStore;
+
+  @Inject
+  public AppFabricClient(AppFabricHttpHandler httpHandler, LocationFactory locationFactory,
+                         AppLifecycleHttpHandler appLifecycleHttpHandler,
+                         ProgramLifecycleHttpHandler programLifecycleHttpHandler,
+                         NamespaceHttpHandler namespaceHttpHandler,
+                         NamespaceAdmin namespaceAdmin, StreamAdmin streamAdmin,
+                         StreamMetaStore streamMetaStore) {
     this.httpHandler = httpHandler;
-    this.serviceHttpHandler = serviceHttpHandler;
     this.locationFactory = locationFactory;
+    this.appLifecycleHttpHandler = appLifecycleHttpHandler;
+    this.programLifecycleHttpHandler = programLifecycleHttpHandler;
+    this.namespaceHttpHandler = namespaceHttpHandler;
+    this.namespaceAdmin = namespaceAdmin;
+    this.streamAdmin = streamAdmin;
+    this.streamMetaStore = streamMetaStore;
   }
 
-  public void reset() {
+  public void reset() throws Exception {
     MockResponder responder = new MockResponder();
     String uri = String.format("/v2/unrecoverable/reset");
     HttpRequest request = new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.PUT, uri);
     httpHandler.resetCDAP(request, responder);
     verifyResponse(HttpResponseStatus.OK, responder.getStatus(), "Reset application failed");
-  }
 
-  public void startProgram(String appId, String flowId, ProgramType type, Map<String, String> args) {
+    // delete all namespaces
+    for (NamespaceMeta namespaceMeta : namespaceAdmin.listNamespaces()) {
+      if (!Constants.DEFAULT_NAMESPACE.equals(namespaceMeta.getName()) &&
+        !Constants.SYSTEM_NAMESPACE.equals(namespaceMeta.getName())) {
+        Id.Namespace namespace = Id.Namespace.from(namespaceMeta.getName());
+        streamAdmin.dropAllInNamespace(namespace);
+        namespaceHttpHandler.deleteDatasets(null, new MockResponder(), namespaceMeta.getName());
```

Comment:
instead of passing null for the request, does it make sense to be consistent with other methods in this class?
`new DefaultHttpRequest(HttpVersion.HTTP_1_1, HttpMethod.POST, uri)`